openedx/core/djangoapps/content/search/handlers.py

```diff
@@ -0,0 +1,35 @@
+"""
+Signal/event handlers for content search
+"""
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+from openedx_events.content_authoring.data import ContentLibraryData, CourseData
+from openedx_events.content_authoring.signals import CONTENT_LIBRARY_CREATED, CONTENT_LIBRARY_DELETED, COURSE_CREATED
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.content.search.models import SearchAccess
+
+
+@receiver(COURSE_CREATED)
+def create_course_search_access(course: CourseData, **kwargs):
```

Comment:
Should these handlers be gated by `api.only_if_meilisearch_enabled` like in https://github.com/openedx/edx-platform/pull/34391/files#diff-0e43f240506a490b34aa601b067fd68ee73118a8158b5941133fe1f807cd47b6R31, to prevent this PR from affecting users not using this feature yet?

Reply:
I think this is a good idea @pomegranited.

Reply:
Will do this after [edx-platform#34391](https://github.com/openedx/edx-platform/pull/34391/files#diff-0e43f240506a490b34aa601b067fd68ee73118a8158b5941133fe1f807cd47b6R31) merges.

Reply:
Yes, but do we even need this handler? You have `get_or_create` in `_meili_access_id_from_context_key` so is there any advantage to pre-creating them here?

Reply:
Good point about the creation handlers, see [a28f3a9](https://github.com/openedx/edx-platform/pull/34471/commits/a28f3a93270cf7f79a8e04eb3924e7b567c195a6).

I left the deletion handlers in so that deleted courses/libraries don't take up any of the 1_000 slots.

Reply:
> I left the deletion handlers in so that deleted courses/libraries don't take up any of the 1_000 slots.

Smart; I didn't think about that.